#!/bin/bash -l
#PBS -l nodes=1:ppn=8,walltime=48:00:00,mem=40gb
#PBS -m abe
#PBS -M durwa004@umn.edu
#PBS -q mcqueue

source ~/.genomestrip_bashrc

cd /home/mccuem/shared/Projects/HorseGenomeProject/Data/ibio_EquCab3/Structural_variants/genomestrip/References/

runDir=/home/mccuem/shared/Projects/HorseGenomeProject/Data/ibio_EquCab3/Structural_variants/genomestrip/References/

readLength=125 
reference=GCF_002863925.1_EquCab3.0_genomic.fasta

export LD_LIBRARY_PATH=${SV_DIR}/bwa:${LD_LIBRARY_PATH}
#These executables must be on your path.
which java > /dev/null || exit 1
which bwa > /dev/null || exit 1

localReference=${runDir}/`echo ${reference} | awk -F / '{ print $NF }'`

#Create svmask for each chromosome
mkdir work
chroms=`cat ${reference}.fai | cut -f 1`

java -Xmx2g -cp ${SV_CLASSPATH} \
    org.broadinstitute.gatk.queue.QCommandLine \

for chr in ${chroms}; do java -Xmx2g -cp ${SV_CLASSPATH} -Xmx4g org.broadinstitute.sv.apps.ComputeGenomeMask -R ${localReference} -O ${runDir}/work/svmask_${chr}.fasta -readLength ${readLength} -sequence ${chr}; done

#Concatenate chromosomes back again:
#Need to run concatenate_chr_fastas.py to get the shell script
#This doesn't seem to work as part of job script
#cd work/
#sh /home/mccuem/shared/Projects/HorseGenomeProject/scripts/EquCab3/ToolsforPMinthehorse/CNV_calling/concat_chrom_fastas.sh

#Index
#cd ../
java -cp ${SV_CLASSPATH} -Xmx4g \
    org.broadinstitute.sv.apps.IndexFastaFile \
    -I GCF_002863925.1_EquCab3.0_genomic.svmask.fasta \
    -O GCF_002863925.1_EquCab3.0_genomic.svmask.fasta.fai

${SV_DIR}/bwa/bwa index -a bwtsw GCF_002863925.1_EquCab3.0_genomic.svmask.fasta 
